components/sidebar.jsx

```diff
@@ -357,7 +354,21 @@ export default class Sidebar extends React.Component {
     }
 
     getDisplayedChannels = () => {
-        return this.state.favoriteChannels.concat(this.state.publicChannels).concat(this.state.privateChannels).concat(this.state.directAndGroupChannels);
+        return getDisplayedChannelsForState(this.state);
+    }
+
+    getDisplayedChannelsForState = (state) => {
+        return state.favoriteChannels.concat(state.publicChannels).concat(state.privateChannels).concat(state.directAndGroupChannels);
+    }
+
+    stateDisplaysChannelId = (state, id) => {
```

Comment:
Could this be called something like `channelIdExistsInState` or similar, just to make its meaning more clear? Based on the current name, I couldn't guess what it was intended to do or what its side effects might be when reading the code above.

Reply:
My problem with "channelIdExistsInState" is that it makes the assumption that all channels existing in the state are displayed. But the state may contain properties that contain undisplayed channels.

Reply:
Agreed, but from the looks of this check, you're just attempting to determine which state (current state vs previous state) the channel id exists in, right? So the method itself is only concerned with figuring out if the specified id is in the specified state, not if the specified id refers to a channel that isn't currently visible.

Reply:
How about `channelIdIsDisplayedForState`? That read any better to you?

Reply:
Yup, that's a definite improvement.